src/derivative_operators/BC_operators.jl

```diff
@@ -37,24 +37,24 @@ end
   The non identity part of Qa is qa:= -b`₁/b0 = -β.*s[2:end]/(α+β*s[1]/Δx). The constant part is Qb = γ/(α+β*s[1]/Δx)
   do the same at the other boundary (amounts to a flip of s[2:end], with the other set of boundary coeffs)
 """
-struct RobinBC{T} <: AffineBC{T}
-    a_l::Vector{T}
+struct RobinBC{T, V<:AbstractVector{T}} <: AffineBC{T}
+    a_l::V
     b_l::T
-    a_r::Vector{T}
+    a_r::V
     b_r::T
-    function RobinBC(l::AbstractVector{T}, r::AbstractVector{T}, dx::T, order = 1) where {T}
+    function RobinBC(l::NTuple{3,T}, r::NTuple{3,T}, dx::T, order = 1) where {T}
         αl, βl, γl = l
         αr, βr, γr = r
 
         s = calculate_weights(1, one(T), Array(one(T):convert(T,order+1))) #generate derivative coefficients about the boundary of required approximation order
 
-        a_l = -s[2:end]./(αl*dx/βl + s[1])
-        a_r = s[end:-1:2]./(αr*dx/βr - s[1]) # for other boundary stencil is flippedlr with *opposite sign*
+        a_l = SVector{length(s)-1}(-s[2:end]./(αl*dx/βl + s[1]))
```

Comment:
this won't infer

Reply:
Can you think of a way to get it to infer?

Reply:
it just won't like these. This constructor should make vectors. If we made the Fornberg function pure then it could.